<?xml version="1.0"?>
<launch>
    <arg name="sim" value="true"/>
    <arg name="gui" default="false"/>
    <arg name="reverse_omega" default="false"/>
    <arg name="use_fake_local_planner" default="true" />

    <include file="$(find docking_sim)/launch/link.launch">
        <arg name="gui" value="$(arg gui)"/>
        <arg name="rviz_path" value="$(find rap_controller)/rviz/rap_planner.rviz"/>
        <arg name="use_fake_local_planner" value="$(arg use_fake_local_planner)"/>
    </include>
    
    <node pkg="rap_controller" type="sim_tf_to_theta.py" name="tf2theta" output='screen'/>
    <node pkg="rap_controller" type="rap_planner.py" name="rap_planner" output='screen'>
        <!-- Kinematic -->
        <param name="crab_kp_vel" value="0.5"/>
        <param name="diff_kp_vel" value="0.5"/>
        <param name="rota_kp_vel" value="1"/> <!-- radian/s-->
        <param name="look_ahead_dist" value="0.8"/>
        <!-- <param name="goal_tolerance_xy" value="0.1"/> -->
        <!-- <param name="goal_tolerance_t" value="10"/> degree -->
        <param name="aside_goal_ang" value="60"/> <!-- degree-->
        
        <!-- System -->
        <param name="ctl_frequency" value="30"/>
        <param name="sim" value="$(arg sim)"/>
        <param name="reverse_omega" value="$(arg reverse_omega)"/>
        <param name="ignore_heading" value="false"/>
        <param name="mode_switch_latch" value="2"/> <!--sec-->
        
        <!-- Tf Frame-->
        <param name="map_frame" value="map"/>
        <param name="map_peer_frame" value="map"/>
        <param name="big_car_frame" value="base_link"/>
        <param name="big_car_peer_frame" value="base_link"/>
        <param name="base_link_frame" value="car1"/>
        <param name="base_peer_frame" value="car2"/>
        <!-- Topic -->
        <param name="global_path_topic" value="/move_base/GlobalPlanner/plan"/>
        <param name="cmd_vel_topic_leader" value="/cmd_vel_car1"/>
        <param name="cmd_vel_topic_follower" value="/cmd_vel_car2"/>
        <param name="goal_topic" value="/move_base_simple/goal"/>
        <param name="costmap_topic" value="/move_base/local_costmap/costmap"/>
    </node>
</launch>
